modules/qna/src/backend/storage.ts

```diff
@@ -119,23 +121,38 @@ export default class Storage {
     }
   }
 
-  async insert(qna: QnaEntry | QnaEntry[]): Promise<string[]> {
-    const ids = await Promise.mapSeries(_.isArray(qna) ? qna : [qna], async (data, i) => {
-      const id = makeID(data)
-      await this.checkForDuplicatedQuestions(data, id)
-      const item: QnaItem = { id, data }
-      if (data.enabled) {
-        await this.createNLUIntentFromQnaItem(item, true)
-      }
-
-      await this.bp.ghost
-        .forBot(this.botId)
-        .upsertFile(this.config.qnaDir, `${id}.json`, JSON.stringify(item, undefined, 2))
+  async upsertItem(item: QnaItem | QnaItem[]): Promise<string[]> {
+    const items = _.isArray(item) ? item : [item]
+    const qnaMap: { [key: string]: QnaEntry } = {}
 
-      return id
+    items.forEach(async item => {
+      if (item.id in qnaMap) {
+        this.bp.logger.warn(`Duplicate IDs found in input while batch importing: ${item.id}`)
+        qnaMap[makeID(item.data)] = item.data
+      }
+      qnaMap[item.id] = item.data
     })
 
-    return ids
+    return Promise.all(
```

Comment:
Can you use `Promise.map(Object.entries(qnaMap), async () => {...})` just as a matter of code consistency